File: plasticcreditledger/wasmbinding/bindings/msg.go
package bindings

type IntertxMsg struct {
	RegisterAccount *RegisterAccount `json:"register_account,omitempty"`
	SetTheString    *SetTheString    `json:"set_the_string,omitempty"`
}

type RegisterAccount struct {
	Owner        string `json:"owner"`
	ConnectionId string `json:"connection_id"`
}

type SetTheString struct {
	TheString string `json:"the_string"`
	Creator   string `json:"creator"`
}

File: cosmwasm/packages/bindings/src/msg.rs
use schemars::JsonSchema;
use serde::{Deserialize, Serialize};

use cosmwasm_std::{CosmosMsg, CustomMsg};

#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
#[serde(rename_all = "snake_case")]
/// A number of Custom messages that can call into the Plastic bindings
pub enum PlasticMsg {
    RegisterAccount {
        owner: String,
        connection_id: String,
    },
    SetTheString {
        the_string: String,
        creator: String,
    },
    SubmitTx {
        owner: String,
        connection_id: String,
        msg: String,
    }
}

impl From<PlasticMsg> for CosmosMsg<PlasticMsg> {
    fn from(msg: PlasticMsg) -> CosmosMsg<PlasticMsg> {
        CosmosMsg::Custom(msg)
    }
}

impl CustomMsg for PlasticMsg {}